app/controllers/invoice_items_controller.rb

```diff
@@ -0,0 +1,14 @@
+class InvoiceItemsController < ApplicationController
+  before_action :get_merchant
+
+  def update
+    invoice_item = InvoiceItem.find(params[:id])
+    invoice_item.update(status: params[:invoice_item][:status])
+    invoice = Invoice.find(invoice_item.invoice_id)
+    redirect_to("/merchants/#{@merchant.id}/invoices/#{invoice.id}")
+  end
+
+  def get_merchant
+    @merchant = Merchant.find(params[:merchant_id])
+  end
```

Comment:
Nice use of the before action!  I like it!  You could put `private` above this method.